src/app/modules/verification/__tests__/verification.controller.spec.ts

```diff
@@ -799,4 +802,152 @@ describe('Verification controller', () => {
       expect(mockRes.json).toHaveBeenCalledWith({ message: expect.any(String) })
     })
   })
+
+  describe('handleResetFieldVerification', () => {
+    const MOCK_REQ = expressHandler.mockRequest({
+      params: {
+        transactionId: MOCK_TRANSACTION_ID,
+        fieldId: MOCK_FIELD_ID,
+        formId: MOCK_FORM_ID,
+      },
+    })
+
+    beforeEach(() =>
+      MockFormService.retrieveFormById.mockReturnValue(okAsync({})),
+    )
+
+    it('should correctly call service when params are valid', async () => {
+      // Arrange
+      MockVerificationFactory.resetFieldForTransaction.mockReturnValueOnce(
+        okAsync(mockTransaction),
+      )
+
+      // Act
+      await VerificationController.handleResetFieldVerification(
+        MOCK_REQ,
+        mockRes,
+        jest.fn(),
+      )
+
+      // Assert
+      expect(
+        MockVerificationFactory.resetFieldForTransaction,
+      ).toHaveBeenCalledWith(MOCK_TRANSACTION_ID, MOCK_FIELD_ID)
+      expect(mockRes.sendStatus).toHaveBeenCalledWith(StatusCodes.OK)
+    })
+
+    it('should return 400 when transaction has expired', async () => {
+      // Arrange
+      MockVerificationFactory.resetFieldForTransaction.mockReturnValueOnce(
+        errAsync(new TransactionExpiredError()),
+      )
+
+      // Act
+      await VerificationController.handleResetFieldVerification(
+        MOCK_REQ,
+        mockRes,
+        jest.fn(),
+      )
+
+      // Assert
+      expect(
+        MockVerificationFactory.resetFieldForTransaction,
+      ).toHaveBeenCalledWith(MOCK_TRANSACTION_ID, MOCK_FIELD_ID)
+      expect(mockRes.status).toHaveBeenCalledWith(StatusCodes.BAD_REQUEST)
+      expect(mockRes.json).toHaveBeenCalledWith({
+        message: expect.any(String),
+      })
+    })
+
+    it('should return 404 when form is not found', async () => {
+      // Arrange
+      MockFormService.retrieveFormById.mockReturnValue(
+        errAsync(new FormNotFoundError()),
+      )
+
+      // Act
+      await VerificationController.handleResetFieldVerification(
+        MOCK_REQ,
+        mockRes,
+        jest.fn(),
+      )
+
+      // Assert
+      expect(
```

Comment:
should we expect that MockFormService was called correctly?

Reply:
i agree we should! fixed in [here](https://github.com/opengovsg/FormSG/pull/1900/commits/a2a7d2f0a51285312dd20fcf42bbcb54b12e73fc)